src/devices/navigation2DClient/backup/Navigation2DClient.cpp

```diff
@@ -0,0 +1,521 @@
+using namespace yarp::dev::Nav2D;
+using namespace yarp::os;
+using namespace yarp::sig;
+
+
+bool Navigation2DClient::set_current_goal_name(const std::string& name)
+{
+    m_current_goal_name = name;
+    return true;
+}
+
+bool Navigation2DClient::get_current_goal_name(std::string& name)
+{
+    if (m_current_goal_name == "")
+    {
+        return false;
+    }
+    name = m_current_goal_name;
+    return true;
+}
+
+bool Navigation2DClient::reset_current_goal_name()
+{
+    m_current_goal_name = "";
+    return true;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------
+
+bool Navigation2DClient::open(yarp::os::Searchable &config)
+{
+    m_local_name.clear();
+    m_navigation_server_name.clear();
+    m_map_locations_server_name.clear();
+    m_localization_server_name.clear();
+
+    m_local_name = config.find("local").asString();
+    m_navigation_server_name = config.find("navigation_server").asString();
+    m_map_locations_server_name = config.find("map_locations_server").asString();
+    m_localization_server_name = config.find("localization_server").asString();
+
+    if (m_local_name == "")
+    {
+        yError("Navigation2DClient::open() error you have to provide a valid 'local' param");
+        return false;
+    }
+
+    if (m_navigation_server_name == "")
+    {
+        yError("Navigation2DClient::open() error you have to provide a valid 'navigation_server' param");
+        return false;
+    }
+
+    if (m_map_locations_server_name == "")
+    {
+        yError("Navigation2DClient::open() error you have to provide valid 'map_locations_server' param");
+        return false;
+    }
+
+    if (m_localization_server_name == "")
+    {
+        yError("Navigation2DClient::open() error you have to provide valid 'localization_server' param");
+        return false;
+    }
+
+    if (config.check("period"))
+    {
+        m_period = config.find("period").asInt32();
+    }
+    else
+    {
+        m_period = 10;
+        yWarning("Navigation2DClient: using default period of %d ms" , m_period);
+    }
+
+    std::string
+            local_rpc_1,
+            local_rpc_2,
+            local_rpc_3,
+            local_rpc_4,
+            remote_rpc_1,
+            remote_rpc_2,
+            remote_rpc_3,
+            remote_streaming_name,
+            local_streaming_name;
+
+    local_rpc_1           = m_local_name           + "/navigation/rpc";
+    local_rpc_2           = m_local_name           + "/locations/rpc";
+    local_rpc_3           = m_local_name           + "/localization/rpc";
+    local_rpc_4           = m_local_name           + "/user_commands/rpc";
+    remote_rpc_1          = m_navigation_server_name + "/rpc";
+    remote_rpc_2          = m_map_locations_server_name + "/rpc";
+    remote_rpc_3          = m_localization_server_name + "/rpc";
+    remote_streaming_name = m_localization_server_name + "/stream:o";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'remote_streaming_name' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5162834)